### PR TITLE
Change how pip is upgraded in ansible.

### PR DESCRIPTION
**Pull request by manthey**

### CODE REVIEW

File: devops/ansible-role-girder/tasks/main.yml [hunk 1]
```diff
@@ -25,8 +25,8 @@
 
 - name: Upgrade pip in the virtualenv
   pip:
-    name: pip
-    extra_args: "--update"
+    name: "pip>=21.0.1"
+    state: present
```
Comment (brianhelba) [inline devops/ansible-role-girder/tasks/main.yml:29]:
```suggestion
    name: pip
    version: ">=21.0.1"
```

File: devops/ansible-role-girder/tasks/main.yml [hunk 2]
```diff
@@ -25,8 +25,8 @@
 
 - name: Upgrade pip in the virtualenv
```
Comment (brianhelba) [inline devops/ansible-role-girder/tasks/main.yml:26]:
```suggestion
- name: Install a recent pip version
```

File: devops/ansible-role-girder/tasks/main.yml [hunk 3]
```diff
@@ -26,7 +26,8 @@
...
   pip:
     name: pip
-    extra_args: "--update"
+    version: ">=21.0.1"
```
Comment (brianhelba) [inline devops/ansible-role-girder/tasks/main.yml:29]: Is this requirement documented anywhere else?

File: devops/ansible-role-girder/tasks/main.yml [hunk 4]
```diff
@@ -23,10 +23,11 @@
 - import_tasks: nodejs.yml
   when: girder_web|bool
 
-- name: Upgrade pip in the virtualenv
+- name: Install a recent pip version
   pip:
     name: pip
-    extra_args: "--update"
+    version: ">=21.0.1"
+    state: present
```
Comment (brianhelba) [inline devops/ansible-role-girder/tasks/main.yml:30]:
```suggestion
```

This is the default.

Comment (manthey) [reply devops/ansible-role-girder/tasks/main.yml:30]: Yep.  You asked for it, though.

File: devops/ansible-role-girder/tasks/main.yml (hunk 3 above)
Comment (manthey) [reply devops/ansible-role-girder/tasks/main.yml:29]: We don't document that we pull in cryptography, and it is a requirement cause by the cryptography module.  How do you recommend we document that an upstream package requires a new version?  We didn't document that we needed pip 9 before this, either (nor check for it).

File: devops/ansible-role-girder/tasks/main.yml (hunk 4 above)
Comment (brianhelba) [reply devops/ansible-role-girder/tasks/main.yml:30]: Yeah, if you're using `state: upgrade`, it's necessary, but the minimum version is better for idempotency. Sorry for the confusion from my lack of specificity.

I have a number of CD pipelines using this role, and minimizing unnecessary Ansible changes is an important requirement.

File: devops/ansible-role-girder/tasks/main.yml (hunk 3 above)
Comment (brianhelba) [reply devops/ansible-role-girder/tasks/main.yml:29]: How about here: https://github.com/girder/girder/blob/master/docs/dependencies.rst ?

**Review by brianhelba — APPROVED** [review]
Great, thanks!

**Pull request merged**